File: DTO/System/Menu/Models/MenuModel.cs
using DTO.Base;

namespace DTO.System.Menu.Models;

public class MenuModel : ModelBase
{
    public string? ControllerName { get; set; }
    public string? Controller { get; set; }
    public string? Action { get; set; }
    public string? Name { get; set; }
    public string? Icon { get; set; }
    public Guid GroupPermissionId { get; set; }
    public string? GroupName { get; set; }
    public int GroupSort { get; set; }
    public bool HasView { get; set; } = false;
    public bool HasAdd { get; set; } = false;
    public bool HasEdit { get; set; } = false;
    public bool HasDelete { get; set; } = false;
    public bool HasApprove { get; set; } = false;
    public bool HasStatistic { get; set; } = false;
    public bool IsShowMenu { get; set; } = true;
}

File: GUI/src/app/models/system/menu.model.ts
import {ToFormControls} from "../base/form-group.model";
import {FormControl, FormGroup, Validators} from "@angular/forms";
import * as uuid from "uuid";
import {GetListRequestModel} from "../base/get-list-request.model";

export type MenuForm = ToFormControls<MenuModel>;

export class MenuModel {
  id: string | null = null;
  controllerName: string | null = null;
  controller: string | null = null;
  action: string | null = null;
  name: string | null = null;
  icon: string | null = null;
  groupPermissionId: string | null = null;
  groupName: string | null = null;
  groupSort: number = 0;
  hasView: boolean = false;
  hasAdd: boolean = false;
  hasEdit: boolean = false;
  hasDelete: boolean = false;
  hasApprove: boolean = false;
  hasStatistic: boolean = false;
  isShowMenu: boolean = true;
  isActive: boolean = true;
  sort: number = 0;
}

export class GetListMenuRequestModel extends GetListRequestModel {
  groupPermissionId: string | null = null;
}

export function createDefaultMenuForm() {
  return new FormGroup<MenuForm>(<MenuForm>{
    id: new FormControl(uuid.v4(), {validators: [Validators.required]}),
    controllerName: new FormControl('', {validators: [Validators.required]}),
    controller: new FormControl('', {validators: [Validators.required]}),
    action: new FormControl('', {validators: [Validators.required]}),
    name: new FormControl('', {validators: [Validators.required]}),
    icon: new FormControl(''),
    groupPermissionId: new FormControl('', {validators: [Validators.required]}),
    hasView: new FormControl(false),
    hasAdd: new FormControl(false),
    hasEdit: new FormControl(false),
    hasDelete: new FormControl(false),
    hasApprove: new FormControl(false),
    hasStatistic: new FormControl(false),
    isShowMenu: new FormControl(true),
    isActive: new FormControl(true),
    sort: new FormControl(0),
  });
}
